src/master-playlist-controller.js

```diff
@@ -624,6 +679,50 @@ export class MasterPlaylistController extends videojs.EventTarget {
   }
 
   /**
+   * Returns the subtitle group for the currently active primary
+   * media playlist.
+   */
+  activeSubtitleGroup_() {
+    let videoPlaylist = this.masterPlaylistLoader_.media();
+    let result;
+
+    if (!videoPlaylist) {
+      return null;
+    }
+
+    if (videoPlaylist.attributes && videoPlaylist.attributes.SUBTITLES) {
+      result = this.subtitleGroups_.groups[videoPlaylist.attributes.SUBTITLES];
+    }
+
+    return result || this.subtitleGroups_.groups.main;
+  }
+
+  activeSubtitleTrack_() {
+    for (let trackName in this.subtitleGroups_.tracks) {
+      if (this.subtitleGroups_.tracks[trackName].mode === 'showing') {
+        return this.subtitleGroups_.tracks[trackName];
+      }
+    }
+
+    return null;
+  }
+
+  handleSubtitleError_() {
+    videojs.log.warn('Problem encountered loading the subtitle track' +
+                     '. Switching back to default.');
+
+    this.subtitleSegmentLoader_.abort();
+
+    let track = this.activeSubtitleTrack_();
+
+    if (track) {
+      track.mode = 'disabled';
+    }
+
+    this.setupSubtitles();
```

Comment:
It shouldn't since just above we set the current active tracks `mode` to `'disabled'`. This has the same effect as turning off subtitles through the UI.